sorting/wiggle_sort.cpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * \addtogroup sorting Sorting Algorithms
+ * @{
+ * \file
+ * \brief [Wiggle Sort Algorithm]
+ * (https://khanhng.com/post/wiggle-sort/) Implementation
+ *
+ * \author [Roshan Kanwar](http://github.com/roshan0708)
+ *
+ * \details
+ * Wiggle Sort sorts the array into a wave like array.
+ * An array ‘arr[0..n-1]’ is sorted in wave form,
+ * if arr[0] >= arr[1] <= arr[2] >= arr[3] <= arr[4] >= …..
+ *
+ * \example
+ * arr = [1,1,5,6,1,4], after wiggle sort arr will become equal to [1,1,6,1,5,4]
+ * arr = [2,8,9,1,7], after wiggle sort arr will become equal to [8,2,9,1,7]
+ */
+
+#include <iostream>  /// for io operations
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+    /**
+    * @namespace wiggle_sort
+    * @brief Functions for [Wiggle Sort](https://khanhng.com/post/wiggle-sort/) algorithm
+    */
+    namespace wiggle_sort {
+        /**
+        * @brief contains various utility functions for completion of the wiggle sort algorithm
+        */
+        class Algorithm {
```

Comment:
is this class even needed? there are no private variables.
The only related function - `wiggleSort` - can be a function in the `sorting::wiggle_sort` namespace. 
the function `display_elements` can be a static function in this code outside of the two namespaces.